test/e2e/performance/perf_test.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package perf
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ksapi "github.com/kubestellar/kubestellar/api/control/v1alpha1"
+	"github.com/kubestellar/kubestellar/test/util"
+)
+
+const (
+	ns = "nginx"
+)
+
+var _ = ginkgo.Describe("performance tests", func() {
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		util.CleanupWDS(ctx, wdsClient, ksWdsClient, ns)
+	})
+
+	ginkgo.Context("performance", func() {
+		ginkgo.It("propagates a new deployment from WDS to WEC", func(ctx context.Context) {
+			const numDeployments = 10
+			sumBinding, sumManifestwork, sumDeploymentWDS, sumDeploymentWEC, sumTotal := 0, 0, 0, 0, 0
+
+			for i := 0; i < numDeployments; i++ {
+				util.CreateBindingPolicy(ctx, ksWdsClient, "performance",
+					[]metav1.LabelSelector{
+						{MatchLabels: map[string]string{"name": "cluster1"}},
+					},
+					[]ksapi.DownsyncPolicyClause{
+						{DownsyncObjectTest: ksapi.DownsyncObjectTest{
+							ObjectSelectors: []metav1.LabelSelector{
+								{MatchLabels: map[string]string{"app.kubernetes.io/name": "nginx"}},
+							}}}})
+				start := time.Now()
```

Comment:
Is it wise to start the timer right after the creation of the BindingPolicy? Creation of a BindingPolicy will trigger some background processing in controllers, which will affect the performance of the measured stuff. I think that the results would be clearer if either (a) the test driver waits for all the background processing to finish before starting the timer or (b) includes the creation of the BindingPolicy in the measured stuff. And (a) looks pretty problematic to me --- how would the test driver know when all the background procesing is done?

Reply:
I adopted (b).

Reply:
So the README and the description in the opening comment of this PR should be updated to reflect the new test design.

But the same issue is present, just at a different boundary.  One iteration of the test is started right after the last delete operation of the previous iteration completes.

Reply:
With a unique deployment and bindingpolicy names, are you still concerned about the delete?

Reply:
Yes. The issue is that the asynchronous processing triggered by the delete at the end of one iteration will still be going on for some prefix of the next iteration.